src/core/echarts.ts

```diff
@@ -605,10 +606,15 @@ class ECharts extends Eventful<ECEventDefinition> {
      * @param opts.replaceMerge Default undefined.
      */
     // Expose to user full option.
-    setOption<Opt extends ECBasicOption>(option: Opt, notMerge?: boolean, lazyUpdate?: boolean): void;
+    setOption<Opt extends ECBasicOption>(
+        option: Opt,
+        notMerge?: boolean,
+        lazyUpdate?: boolean,
+        immutableMode?: boolean
+      ): void;
     setOption<Opt extends ECBasicOption>(option: Opt, opts?: SetOptionOpts): void;
     /* eslint-disable-next-line */
-    setOption<Opt extends ECBasicOption>(option: Opt, notMerge?: boolean | SetOptionOpts, lazyUpdate?: boolean): void {
+    setOption<Opt extends ECBasicOption>(option: Opt, notMerge?: boolean | SetOptionOpts, lazyUpdate?: boolean, immutableMode?: boolean): void {
```

Comment:
I definitely prefer named params using `opt` over passing separate booleans (this seems to still be around for backwards compatibility), but I added both for now since I was following the existing pattern